website/docs/d/resource.html.markdown

```diff
@@ -0,0 +1,44 @@
+---
+layout: "kubernetes"
+page_title: "Kubernetes: kubernetes_resource"
+description: |-
+  This is a generic data source for Kubernetes API resources
+---
+
+# kubernetes_resource
+
+This data source is a generic way to retrieve resources from the Kubernetes API. 
+
+### Example: Get data from a ConfigMap
+
+```hcl
+data "kubernetes_resource" "example" {
+  apiVersion = "v1"
```

Comment:
Worth noting that we're going to break the `snake_case` convention that is used everywhere else (and all other providers too) for resource attributes. `api_version` would be fine here, just that the camelCase convention is probably more familiar to users of this provider.

Reply:
I honestly am ok with either option. There are valid points, which you raised, for both of them.
I'm with you there that keeping with the spirit of this resources trying to track as close as possible to native K8s, this looks like the slightly better option.

Reply:
I've changed this to `api_version` after my conversation with DevEx – they recommended not breaking the convention.